### PR TITLE
Add query assist data summary agent template

### PR DESCRIPTION
**Pull request by chishui**
### Description
Add query assist data summary agent template

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/flow-framework/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sample-templates/query-assist-data-summary-agent-claude-tested.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                  "request_body": "{\"prompt\":\"${parameters.prompt}\", \"max_tokens_to_sample\":${parameters.max_tokens_to_sample}, \"temperature\":${parameters.temperature},  \"anthropic_version\":\"${parameters.anthropic_version}\" }",
+                  "action_type": "predict",
+                  "url": "https://bedrock-runtime.us-west-2.amazonaws.com/model/anthropic.claude-instant-v1/invoke"
+                }
+              ],
+              "credential": {
+                  "access_key": "<YOUR_ACCESS_KEY>",
+                  "secret_key": "<YOUR_SECRET_KEY>",
+                  "session_token": "<YOUR_SESSION_TOKEN>"
+              },
```
Comment (dbwiddis) [inline sample-templates/query-assist-data-summary-agent-claude-tested.json:38]: Not a comment on this PR, but note to other reviewers. I wonder if there's a way we can (probably in ML Commons) not require these to be specified in the connector, but instead have them pulled in real time using CredentialsProviders like the AWS SDK client does.

**Review by dbwiddis — COMMENTED** [review]
LGTM. Please add a change log entry!

**Comment by chishui** [comment]
> LGTM. Please add a change log entry!

Added

**Review by dbwiddis — APPROVED** [review]

**Comment by chishui** [comment]
@dbwiddis  could you help merge this PR? Thanks

**Comment by dbwiddis** [comment]
> @dbwiddis could you help merge this PR? Thanks

I've approved already. It takes two approvals. :)

We have a while before the next release, there is no rush.

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**